src/System.CommandLine/Binding/ServiceBindingSide.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.CommandLine.Binding
+{
+    public class ServiceBindingSide : BindingSide
+    {
+        private ServiceBindingSide(BindingGetter getter, BindingSetter setter)
+           : base(getter, setter)
+        { }
+
+        public static ServiceBindingSide Create(Type serviceType)
+             => new ServiceBindingSide((c, t) => c.ServiceProvider.GetService(serviceType),  null);
```

Comment:
You should pass `t` rather than `serviceType` to the invocation inside the lambda.

Reply:
I am renaming the lambda parameters because this wasn't clear. t is not the service type.

Reply:
I did this the other places that had the (c,t) for the bind